contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -203,92 +205,118 @@ contract WithdrawalQueue is AccessControlEnumerable, WithdrawalQueueBase {
+    ///  `msg.sender` will be used as recipient.
+    /// @return requestIds an array of the created withdrawal requests
+    function requestWithdrawalsWstETH(WithdrawalRequestInput[] calldata _withdrawalRequestInputs)
+        public
+        whenResumed
+        returns (uint256[] memory requestIds)
+    {
+        requestIds = new uint256[](_withdrawalRequestInputs.length);
+        for (uint256 i = 0; i < _withdrawalRequestInputs.length; ++i) {
+            uint256 amountOfWstETH = _withdrawalRequestInputs[i].amount;
+            address recipient = _checkWithdrawalRequestInput(
+                IWstETH(WSTETH).getStETHByWstETH(amountOfWstETH),
+                _withdrawalRequestInputs[i].recipient
+            );
+            requestIds[i] = _requestWithdrawalWstETH(amountOfWstETH, recipient);
+        }
     }
 
-    /**
-     * @notice Request withdrawal of the provided stETH token amount using EIP-2612 Permit
-     * @param _amountOfStETH StETH tokens that will be locked for withdrawal
-     * @param _recipient address to send ether to upon withdrawal. Will be set to `msg.sender` if `address(0)` is passed
-     */
-    function requestWithdrawalWithPermit(
-        uint256 _amountOfStETH,
-        address _recipient,
-        uint256 _deadline,
-        uint8 _v,
-        bytes32 _r,
-        bytes32 _s
-    ) external whenResumed returns (uint256 requestId) {
-        _recipient = _checkWithdrawalRequestInput(_amountOfStETH, _recipient);
-
-        STETH.permit(msg.sender, address(this), _amountOfStETH, _deadline, _v, _r, _s);
-
-        return _requestWithdrawal(_amountOfStETH, _recipient);
+    struct Permit {
+        uint256 value;
+        uint256 deadline;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
     }
 
-    /**
-     * @notice Request withdrawal of the provided wstETH token amount
-     * @param _amountOfWstETH StETH tokens that will be locked for withdrawal
-     * @param _recipient address to send ether to upon withdrawal. Will be set to `msg.sender` if `address(0)` is passed
-     */
-    function requestWithdrawalWstETH(uint256 _amountOfWstETH, address _recipient)
-        external
-        whenResumed
-        returns (uint256 requestId)
-    {
-        _recipient = _checkWithdrawalRequestInput(IWstETH(WSTETH).getStETHByWstETH(_amountOfWstETH), _recipient);
-        return _requestWithdrawalWstETH(_amountOfWstETH, _recipient);
+    /// @notice Request the sequence of stETH withdrawals according to passed `withdrawalRequestInputs` data using EIP-2612 Permit
+    /// @param _withdrawalRequestInputs an array of `WithdrawalRequestInput` data. The standalone withdrawal request will
+    ///  be created for each item in the passed list. If `WithdrawalRequestInput.recipient` is set to `address(0)`,
+    ///  `msg.sender` will be used as recipient.
+    /// @return requestIds an array of the created withdrawal requests
```

Comment:
permit param is undocumented

Reply:
Added the doc comment for this param though maybe it is not very detailed.